source/compiler/qsc_frontend/src/lower.rs

```diff
@@ -376,10 +380,19 @@ impl With<'_> {
         match hir::Attr::from_str(attr.name.name.as_ref()) {
             Ok(hir::Attr::EntryPoint) => match &*attr.arg.kind {
                 ast::ExprKind::Tuple(args) if args.is_empty() => Some(hir::Attr::EntryPoint),
+                // @EntryPoint(Profile)
+                ast::ExprKind::Paren(inner)
+                    if matches!(inner.kind.as_ref(), ast::ExprKind::Path(PathKind::Ok(path))
+                if Profile::from_str(path.name.name.as_ref()).is_ok()) =>
+                {
+                    Some(hir::Attr::EntryPoint)
+                }
+                // Any other form is not valid so generates an error.
                 _ => {
-                    self.lowerer
-                        .errors
-                        .push(Error::InvalidAttrArgs("()".to_string(), attr.arg.span));
+                    self.lowerer.errors.push(Error::InvalidAttrArgs(
+                        "profile name".to_string(),
```

Comment:
Since `()` is still valid, the error should not imply that profile name is required.

```suggestion
                        "empty or profile name ".to_string(),
```

Reply:
done